crates/router/src/db/merchant_connector_account.rs

```diff
@@ -374,28 +387,80 @@ impl MerchantConnectorAccountInterface for MockDb {
 
     async fn find_merchant_connector_account_by_merchant_id_and_disabled_list(
         &self,
-        _merchant_id: &str,
-        _get_disabled: bool,
+        merchant_id: &str,
+        get_disabled: bool,
     ) -> CustomResult<Vec<domain::MerchantConnectorAccount>, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let accounts = self
+            .merchant_connector_accounts
+            .lock()
+            .await
+            .iter()
+            .filter(|account: &&storage::MerchantConnectorAccount| {
+                account.merchant_id == merchant_id && account.disabled == Some(get_disabled)
+            })
```

Comment:
```suggestion
            .filter(|account: &&storage::MerchantConnectorAccount| {
                if get_disabled {
                    account.merchant_id == merchant_id
                } else {
                    account.merchant_id == merchant_id && account.disabled == Some(false)
                }
            })
```

It would have to be something along these lines to mimic the behavior of the [SQL query that would be executed](https://github.com/juspay/hyperswitch/blob/61bacd8c9590a78a6d5067e378bfed6301d64d07/crates/storage_models/src/query/merchant_connector_account.rs#L89-L125).